crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -332,25 +337,36 @@ impl World {
         I: SystemInput + 'static,
         O: 'static,
     {
+        if !self.is_resource_added::<RunSystemStack>() {
+            self.init_resource::<RunSystemStack>();
+        }
```

Comment:
`init_resource` checks whether the resource already exists, so you can simplify this.

```suggestion
        self.init_resource::<RunSystemStack>();
```